File: gluecodium/src/test/resources/smoke/attributes/output/android/com/example/smoke/AttributesClass.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
@OnClass
public final class AttributesClass extends NativeBase {
    @OnConstInClass
    public static final boolean PI = false;
    /**
     * For internal use only.
     * @hidden
     */
    protected AttributesClass(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @OnFunctionInClass
    public native void veryFun(@OnParameterInClass @NonNull final String param);
    @OnPropertyInClass
    @NonNull
    public native String getProp();
    @OnPropertyInClass
    public native void setProp(@NonNull final String value);
}

File: gluecodium/src/test/resources/smoke/attributes/output/android-kotlin/com/example/smoke/AttributesClass.kt
/*

 *
 */

@file:JvmName("AttributesClassExtensions")

package com.example.smoke

import com.example.NativeBase

@OnClass
class AttributesClass : NativeBase {



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}




    @OnFunctionInClass

    external fun veryFun(@OnParameterInClass param: String) : Unit

    @OnPropertyInClass
    var prop: String
        external get
        external set




    companion object {
        @OnConstInClass
        @JvmField final val PI: Boolean = false
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}

